pkg/acsengine/template_generator.go

```diff
@@ -52,25 +52,22 @@ func (t *TemplateGenerator) GenerateTemplate(containerService *api.ContainerServ
 	defer func() {
 		properties.OrchestratorProfile.OrchestratorVersion = orchVersion
 	}()
-	if certsGenerated, err = setPropertiesDefaults(containerService, isUpgrade, isScale); err != nil {
```

Comment:
So to be clear, we've decomposed defaults enforcement from the main `GenerateTemplate` func? So going forward `GenerateTemplate` will no assume a fully rationalized api model?

Reply:
Yes. The PR goes with that assumption.